mtags/src/main/scala/scala/meta/internal/mtags/ListFiles.scala

```diff
@@ -1,40 +1,15 @@
 package scala.meta.internal.mtags
 
-import java.nio.file.FileVisitResult
-import java.nio.file.Files
-import java.nio.file.NoSuchFileException
-import java.nio.file.Path
-import java.nio.file.SimpleFileVisitor
-import java.nio.file.attribute.BasicFileAttributes
 import scala.meta.io.AbsolutePath
+import scala.collection.JavaConverters._
 
 object ListFiles {
   def foreach(root: AbsolutePath)(fn: AbsolutePath => Unit): Unit = {
     ListFiles(root).foreach(fn)
   }
-  def apply(root: AbsolutePath): Traversable[AbsolutePath] =
-    new Traversable[AbsolutePath] {
-      override def foreach[U](fn: AbsolutePath => U): Unit = {
-        if (root.isFile) fn(root)
-        else if (root.isDirectory) {
-          try {
-            Files.walkFileTree(
-              root.toNIO,
-              new SimpleFileVisitor[Path] {
-                override def visitFile(
-                    file: Path,
-                    attrs: BasicFileAttributes
-                ): FileVisitResult = {
-                  fn(AbsolutePath(file))
-                  super.visitFile(file, attrs)
-                }
-              }
-            )
-          } catch {
-            case _: NoSuchFileException =>
-              () // error is reported by the JDK
-          }
-        }
-      }
+  def apply(root: AbsolutePath): Iterable[AbsolutePath] =
+    new Iterable[AbsolutePath] {
+      override def iterator: Iterator[AbsolutePath] =
```

Comment:
An iterator is not safe to use when walking file trees since we need to properly close the streams provided by NIO. It's better to return a `collection.Seq` like `ArrayBuffer`. It might be worth considering using Haoyi's geny here eventually 🤔